src/main/resources/com/google/api/codegen/java/http_unit_test.snip

```diff
@@ -0,0 +1,139 @@
+      {@testCase(test)}
+
+    @end
+  }
+@end
+
+@private testCase(test)
+  @switch test.clientMethodType
+  @case "FlattenedMethod"
+      {@unaryTestCase(test)}
+  @case "PagedFlattenedMethod"
+      {@unaryTestCase(test)}
+  @case "AsyncOperationFlattenedMethod"
+      {@longRunningMethodTestCase(test)}
+  @end
+@end
+
+@private unaryTestCase(test)
+  @@Test
+  @@SuppressWarnings("all")
+  public void {@test.name}() {
+    {@addResponse(test)}
+
+    {@initCode(test.initCode)}
+
+    {@methodCall(test)}
+
+    {@unarySuccessAsserts(test)}
+  }
+
+  @@Test
+  @@SuppressWarnings("all")
+  public void {@test.nameWithException}() throws Exception {
+    {@addException(test)}
+
+    try {
+      {@initCode(test.initCode)}
+
+      client.{@test.clientMethodName}(\
+          {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+      Assert.fail("No exception raised");
+    } catch (InvalidArgumentException e) {
+      // Expected exception
+    }
+  }
+@end
+
+@private addResponse(test)
+  @if test.hasReturnValue
+    {@initCode(test.mockResponse.initCode)}
+    MOCK_SERVICE.addResponse(expectedResponse);
+  @end
+  {@""}
+  MOCK_SERVICE.setSerializer({@test.methodDescriptor});
+@end
+
+@private addException(test)
+  ApiException exception = ApiExceptionFactory.createException(new Exception(), FakeStatusCode.of(Code.INVALID_ARGUMENT), false);
+  MOCK_SERVICE.addException(exception);
+  MOCK_SERVICE.setSerializer({@test.methodDescriptor});
+@end
+
+@private methodCall(test)
+  @switch test.clientMethodType
+  @case "PagedFlattenedMethod"
+    {@test.responseTypeName} pagedListResponse = client.{@test.clientMethodName}(\
+      {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+
+    @join pageStreamingResponseView : test.pageStreamingResponseViews
+      List<{@pageStreamingResponseView.resourceTypeName}> {@pageStreamingResponseView.resourcesVarName} = Lists.newArrayList(pagedListResponse.{@pageStreamingResponseView.resourcesIterateMethod}());
+      Assert.assertEquals(1, {@pageStreamingResponseView.resourcesVarName}.size());
+      Assert.assertEquals(expectedResponse.{@pageStreamingResponseView.resourcesFieldGetterName}().get(0), {@pageStreamingResponseView.resourcesVarName}.get(0));
+    @end
+  @case "FlattenedMethod"
+    @if test.hasReturnValue
+      {@test.responseTypeName} actualResponse =
+          client.{@test.clientMethodName}(\
+          {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+      Assert.assertEquals(expectedResponse, actualResponse);
+    @else
+      client.{@test.clientMethodName}(\
+            {@sampleMethodCallArgList(test.initCode.fieldSettings)});
+    @end
+  @default
+    $unhandledCase: {@test.clientMethodType}$
+  @end
+@end
+
+@private longRunningMethodCall(test)
+  {@test.responseTypeName} actualResponse =
+      client.{@test.clientMethodName}(\
+          {@sampleMethodCallArgList(test.initCode.fieldSettings)}).get();
+  Assert.assertEquals(expectedResponse, actualResponse);
+@end
```

Comment:
Maybe remove this since it's not supported yet?

Reply:
Sure.